### PR TITLE
Personal account notes

### PR DESCRIPTION
**Pull request by connyduck**
- notes are saved automatically once you stop typing
- note field is only shown when server supports it
- I also cleaned up some of the old `Call` things and replaced them with RxJava `Single`

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/viewmodel/AccountViewModel.kt [hunk 1]
```diff
@@ -243,37 +219,35 @@ class AccountViewModel @Inject constructor(
...
-            RelationShipAction.UNMUTE -> mastodonApi.unmuteAccount(accountId)
-        }
-
-        call.enqueue(callback)
-        callList.add(call)
-
+    fun noteChanged(newNote: String) {
+        noteSaved.postValue(false)
+        noteDisposable?.dispose()
+        noteDisposable = Single.timer(1500, TimeUnit.MILLISECONDS)
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/viewmodel/AccountViewModel.kt:233]: This very much looks like some `throttle`, I think RxAndroid has a way to get a stream from a field?

**Review by charlag — COMMENTED** [review]
Code looks good, would be nice to finally have it!
I didn't try it yet, i will post once I build and try it

File: app/src/main/java/com/keylesspalace/tusky/viewmodel/AccountViewModel.kt (hunk 1 above)
Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/viewmodel/AccountViewModel.kt:233]: You mean something like [TextViewTextChangesObservable](https://github.com/JakeWharton/RxBinding/blob/master/rxbinding/src/main/java/com/jakewharton/rxbinding4/widget/TextViewTextChangesObservable.kt)? It is part of RxBinding which we currently don't have in the project, but I can add it if you think its better.

Comment (charlag) [reply app/src/main/java/com/keylesspalace/tusky/viewmodel/AccountViewModel.kt:233]: Oh, I thought we either had the library or had a home-grown alternative but maybe I'm mistaking it with something

**Comment by charlag** [comment]
I tried it out and it works well! I just thought maybe we should disable the field until we load things to not overwrite by accident?

File: app/src/main/java/com/keylesspalace/tusky/viewmodel/AccountViewModel.kt (hunk 1 above)
Comment (charlag) [reply app/src/main/java/com/keylesspalace/tusky/viewmodel/AccountViewModel.kt:233]: Maybe we should at least modify our `addTextChangedListener` from `ViewExtensions` to allow unsubscribing and use that instead?

**Pull request merged**